backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/service/SpecificationDataParser.kt

```diff
@@ -126,10 +127,12 @@ class SpecificationDataParser @Autowired constructor(
 
     }
 
-    private fun getConsistentServiceId(idFromUpload: String?, idFromPath: String?): ServiceId {
-        if (idFromUpload != null && idFromPath != null && idFromUpload != idFromPath) {
-            throw MismatchedServiceIdException(idFromUpload, idFromPath)
+    private fun getConsistentServiceId(ids: List<String?>): ServiceId {
+        val notNullIds: List<String> = ids.filterNotNull()
```

Comment:
What should happen if you manually enter the same id as the `x-api-id`. In my opinion, it should work.